axelor-supplychain/src/main/resources/views/Mrp.xml

```diff
@@ -80,8 +81,9 @@
 	    <view type="grid" name="mrp-line-grid"/>
 	    <view type="form" name="mrp-line-form"/>
 		<view-param name="limit" value="200"/>
-	    <domain>self.mrp.id = :_mrpId</domain>
+		<domain>self.mrp.id = :_mrpId AND self.product.id IN (select m.product from MrpLine as m where m.mrpLineType.elementSelect in (:_elementSelect))</domain>
 		<context name="_mrpId" expr="eval: __this__.id"/>
+		<context name="_elementSelect" expr="eval: !displayProductWithoutProposal ? [__repo__(MrpLineType).ELEMENT_PURCHASE_PROPOSAL,__repo__(MrpLineType).ELEMENT_MANUFACTURING_PROPOSAL,__repo__(MrpLineType).ELEMENT_MANUFACTURING_PROPOSAL_NEED]: __repo__(MrpLine).all().fetch().collect{it->it.mrpLineType.elementSelect}"/>
```

Comment:
Remove '__repo__(MrpLine).all().fetch().collect{it->it.mrpLineType.elementSelect}' and keep null and change filter accordingly